File: src/util/xml/XMLParse.java
package util.xml;

import java.io.File;
import java.lang.reflect.Field;
import java.util.HashMap;

import javax.xml.parsers.DocumentBuilder;
import javax.xml.parsers.DocumentBuilderFactory;

import org.w3c.dom.Document;
import org.w3c.dom.NamedNodeMap;
import org.w3c.dom.NodeList;

public class XMLParse {
	
	private static String xmlFilePath = "config";
	public static HashMap xmlParseMap = null;
	
	public static void execute() {
		try {
			domParse();
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
	
	protected static void domParse() throws Exception {
		File[] files = new File(xmlFilePath).listFiles();
		for(int i = 0; i < files.length; i++) {
			File xmlFile = new File(files[i].getPath());
						
			DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();   
			DocumentBuilder builder = factory.newDocumentBuilder();   
			Document doc = builder.parse(xmlFile);   
			doc.getDocumentElement().normalize();
			
			putParseMapByXML(doc);
		}
	}
	
	protected static void putParseMapByXML(Document doc) throws Exception {
		Field[] fileds = XMLConstant.class.getFields();
		for(int i = 0; i < fileds.length; i++) {
			Field f = fileds[i];
			String prefix = XMLConstant.PREFIX_TAG + "_";
			if(prefix.indexOf((String)f.get(f.getName())) < 0) {
				NodeList list = doc.getElementsByTagName((String)f.get(f.getName()));
				for(int j = 0; j < list.getLength(); j++) {
					NamedNodeMap map = list.item(j).getAttributes();
					xmlParseMap.put(list.item(j).getNodeName(), map);
				}				
			}
		}
	}
	
	protected static void test() {
		System.out.println("테스트 입니다.");
	}
}

File: src/util/xml/xml_parse.py
import os
import xml.etree.ElementTree as ET
from typing import Dict

from .xml_constant import XMLConstant


class XMLParse:
    """Parse XML configuration files from the config directory."""
    xml_file_path = "config"
    xml_parse_map: Dict[str, Dict[str, str]] = {}

    @classmethod
    def execute(cls) -> None:
        try:
            cls._dom_parse()
        except Exception as e:
            print(e)

    @classmethod
    def _dom_parse(cls) -> None:
        for filename in os.listdir(cls.xml_file_path):
            filepath = os.path.join(cls.xml_file_path, filename)
            if not filepath.lower().endswith(".xml"):
                continue
            doc = ET.parse(filepath)
            root = doc.getroot()
            cls._put_parse_map_by_xml(root)

    @classmethod
    def _put_parse_map_by_xml(cls, root: ET.Element) -> None:
        for name in dir(XMLConstant):
            if name.isupper():
                value = getattr(XMLConstant, name)
                prefix = f"{XMLConstant.PREFIX_TAG}_"
                if not value.startswith(prefix):
                    for elem in root.findall(value):
                        cls.xml_parse_map[elem.tag] = elem.attrib
